contracts/facets/GBMFacet.sol

```diff
@@ -181,6 +178,81 @@ contract GBMFacet is IGBM, IERC1155TokenReceiver, IERC721TokenReceiver, Modifier
         emit Auction_ItemClaimed(_auctionID);
     }
 
+    /// @notice Attribute a token to the caller and distribute the proceeds to the owner of this contract.
+    /// throw if bidding is disabled or if the auction is not finished.
+    /// @param _auctionID The auctionId of the auction to complete
+    //No change necessary for this function code, but it use overriden internal and hence need overriding too in the diamond
+    function buyNow(uint256 _auctionID) public {
+        Auction storage a = s.auctions[_auctionID];
+        if (a.owner == address(0)) revert("NoAuction");
+        if (a.claimed == true) revert("AuctionClaimed");
+
+        address ca = a.tokenContract;
+        uint256 tid = a.info.tokenID;
+        uint256 tam = a.info.tokenAmount;
+        uint256 ae1bnp = a.buyItNowPrice;
+
+        if (ae1bnp == 0) revert("NoBuyItNowPrice");
+        if (((ae1bnp * s.buyItNowInvalidationThreshold) / 100) <= a.highestBid) revert("HighestBidTooHighToBuyNow");
+
+        if (msg.sender == a.owner) revert("OwnerBuyNowNotAllowed");
+        if (a.info.startTime > block.timestamp) revert("AuctionNotStarted");
+        //verify existence
+        if (a.owner == address(0)) revert("NoAuction");
+        if (a.info.endTime < block.timestamp) revert("AuctionEnded");
+        if (a.claimed == true) revert("AuctionClaimed");
+        if (a.biddingAllowed == false) revert("BiddingNotAllowed");
+
+        address tokenContract = a.tokenContract;
+        if (s.contractBiddingAllowed[tokenContract] == false) revert("BiddingNotAllowed");
+
+        //royalties
+        address[] memory royalties;
+        uint256[] memory royaltyShares;
+
+        //Prevents re-entrancy
+        a.claimed = true;
+
+        //Transfer the money of the buyer to the GBM Diamond
+        IERC20(s.GHST).transferFrom(msg.sender, address(this), ae1bnp);
+
+        //Refund the highest bidder
+        if (a.highestBid > 0) {
+            IERC20(s.GHST).transfer(a.highestBidder, a.highestBid + a.dueIncentives);
+            //emit incentive event and bidRemoval event
+            emit Auction_IncentivePaid(_auctionID, a.highestBidder, a.dueIncentives);
+            emit Auction_BidRemoved(_auctionID, a.highestBidder, a.highestBid);
+        }
+
+        emit Auction_BoughtNow(_auctionID, msg.sender);
+
+        if (IERC165(ca).supportsInterface(0x2a55205a)) {
```

Comment:
Please refactor this code so we're not duplicating code (it also appears in the "claim" function) @orionstardust